docs/source/Howtos/Extending-the-REST-API.md

```diff
@@ -0,0 +1,287 @@
+	def inventory(self, request, pk=None):
+		obj = self.get_object()
+		return Response( InventorySerializer(obj).data, status=status.HTTP_200_OK )
+```
+
+That'll use our new serializer to get our character's inventory! Except... not quite. Go ahead and try it: `evennia reboot` and then `/api/characters/1/inventory` like before. Unlike before, you should get an error saying you don't have permission.
+
+## Customizing API permissions
+
+Evennia comes with its own custom API permissions class, connecting the API permissions to the in-game permission hierarchy and locks system. Since we're trying to access the object's data now, we need to pass the `has_object_permission` check as well as the general permission check - and that permission class hardcodes the actions into the object permission checks.
+
+So, since we've added a new action - `inventory` - to our characters endpoint, we need to use our own custom permissions on our characters endpoint. Create one more module file: `<mygame>/web/api/permissions.py`
+
+Like with the previous classes, we'll be inheriting from the original and extending it to take advantage of all the work Evennia already does for us.
+
+```python
+from evennia.web.api.permissions import EvenniaPermission
+
+class CharacterPermission(EvenniaPermission):
+	
+	def has_object_permission(self, request, view, obj):
+		"""
+		Checks object-level permissions after has_permission
+		"""
+		# our new permission check
+		if view.action == "inventory":
+			return self.check_locks(obj, request.user, self.view_locks)
+
+		# if it's not an inventory action, run through all the default checks
+		return super().has_object_permission(request, view, obj)
+```
+
+That's the whole permission class! For our final step, we just need to use it in our characters view by importing it and setting the `permission_classes` property.
+
+Once you've done that, your final `views.py` should look like this:
+
+```python
+"""
+<mygame>/web/api/views.py
+
+Customized views for the REST API
+"""
+# we'll need these from django's rest framework to make our view work
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework import status
+
+# this implements all the basic Evennia Object endpoint logic, so we're inheriting from it
+from evennia.web.api.views import ObjectDBViewSet
+
+# and we need this to filter our character view
+from evennia.objects.objects import DefaultCharacter
+
+from .serializers import InventorySerializer
+from .permissions import CharacterPermission
+
+# our own custom view
+class CharacterViewSet(ObjectDBViewSet):
+	"""
+	A customized Character view that adds an inventory detail
+	"""
+	permission_classes = [CharacterPermission]
+	queryset = DefaultCharacter.objects.all_family()
+
+	@action(detail=True, methods=["get"])
+	def inventory(self, request, pk=None):
+		obj = self.get_object()
+		return Response( InventorySerializer(obj).data, status=status.HTTP_200_OK )
+```
+
```

Comment:
A suggestion is to have a final `## Conclusions` section at the end. This can summarize what you have learned, suggest further things to try and even link to some other tutorial(s) that could be good to try next.